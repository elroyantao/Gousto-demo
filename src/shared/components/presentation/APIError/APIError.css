.Overlay {
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: gray(50);
  opacity: .7;
}

.Modal {
  display: flex;
  flex-direction: column;
  width: 400px;
  max-width: 100%;
  height: 200px;
  max-height: 100%;
  opacity: 1;
  background: white;
}

.Modal-content {
  flex: 1;
  display: flex;
  justify-content: center;
  align-items: center;
}

.Modal-action {
  flex-basis: 50px;
  display: flex;
  justify-content: space-around;
  align-items: center;
  background: gray(180)
}

.Modal-button {
  padding: 10px 20px;
}

.Modal-button--reload {
  composes: Modal-button;
  color: green;
}
.Modal-button--close {
  composes: Modal-button;
  color: red;
}
