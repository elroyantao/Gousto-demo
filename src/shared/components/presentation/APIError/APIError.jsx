import React, { Component } from 'react'
import PropTypes from 'prop-types'

import styles from './APIError.css'

export default class APIError extends Component {
  static propTypes = {
    hasError: PropTypes.bool,
    clearError: PropTypes.func
  }

  reloadSite() {
    window.location.reload()
  }

  render() {
    const { hasError, clearError } = this.props
    if (!hasError) return null
    return (
      <div className={styles.Overlay}>
        <div className={styles.Modal}>
          <div className={styles['Modal-content']}>
            There was an error in the api call
          </div>
          <div className={styles['Modal-action']}>
            <button className={styles['Modal-button--close']} onClick={clearError}>
              Close
            </button>
            <button className={styles['Modal-button--reload']} onClick={this.reloadSite}>
              Reload
            </button>
          </div>
        </div>
      </div>
    )
  }
}
