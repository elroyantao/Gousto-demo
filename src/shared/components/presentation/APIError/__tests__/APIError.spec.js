import testComponentHelper from '../../../../lib/test-helper'
import APIError from '../APIError'

describe('<APIError />', () => {
  const initialState = {
    hasError: false,
    clearError: jest.fn()
  }
  const renderComponent = testComponentHelper(APIError)
  describe('@renders', () => {
    it('in default state', () => {
      expect(renderComponent(initialState).getTree())
        .toMatchSnapshot()
    })
    it('in error state', () => {
      expect(renderComponent({
        ...initialState,
        hasError: true
      }).getTree())
        .toMatchSnapshot()
    })
  })
  describe('@events', () => {
    let renderedComponent
    beforeAll(() => {
      renderedComponent = renderComponent({
        ...initialState,
        hasError: true
      })
    })
    describe('onClick reload button', () => {
      window.location.reload = jest.fn()

      it('should call window.location.reload()', () => {
        const { wrapper } = renderedComponent
        expect(window.location.reload).not.toHaveBeenCalled()
        wrapper.find('.Modal-button--reload').simulate('click')
        expect(window.location.reload).toHaveBeenCalledTimes(1)
      })
    })
    describe('onClick close button', () => {
      it('should call clearError', () => {
        const { instance, wrapper } = renderedComponent
        expect(instance.props.clearError).not.toHaveBeenCalled()
        wrapper.find('.Modal-button--close').simulate('click')
        expect(instance.props.clearError).toHaveBeenCalledTimes(1)
      })
    })
  })
})
