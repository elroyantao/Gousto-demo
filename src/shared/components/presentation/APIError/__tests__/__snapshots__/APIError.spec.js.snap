// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<APIError /> @renders in default state 1`] = `null`;

exports[`<APIError /> @renders in error state 1`] = `
<div
  className="Overlay"
>
  <div
    className="Modal"
  >
    <div
      className="Modal-content"
    >
      There was an error in the api call
    </div>
    <div
      className="Modal-action"
    >
      <button
        className="Modal-button--close"
        onClick={[Function]}
      >
        Close
      </button>
      <button
        className="Modal-button--reload"
        onClick={[Function]}
      >
        Reload
      </button>
    </div>
  </div>
</div>
`;
